OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -500,12 +500,17 @@ bool HasAdequateAirUnitReloadBuildings(ActorInfo actorInfo)
 			if (aircraftInfo == null)
 				return true;
 
-			// If the aircraft has at least 1 AmmoPool and defines 1 or more RearmBuildings, check if we have enough of those
-			var hasAmmoPool = actorInfo.TraitInfos<AmmoPoolInfo>().Any();
-			if (hasAmmoPool && aircraftInfo.RearmBuildings.Count > 0)
+			// If actor isn't Rearmable, it doesn't need a building to reload
+			var rearmableInfo = actorInfo.TraitInfoOrDefault<RearmableInfo>();
+			if (rearmableInfo == null)
+				return true;
+
+			// If the aircraft has at least 1 AmmoPool, check if we have enough RearmActors
+			var hasAmmoPool = actorInfo.HasTraitInfo<AmmoPoolInfo>();
```

Comment:
You only use this variable once, so can move the `HasTraitInfo` inline in the `if`?